client/src/components/ContestDetail.js

```diff
@@ -105,26 +113,43 @@ const submitLink = React.forwardRef((props, ref) => {
   return <Link innerRef={ref} to={`${id}/submit`} {...props} />;
 });
 
-const ContestDetail = ({ classes, auth, contest, match, history }) => {
+const ContestDetail = ({ classes, auth, match, history }) => {
   const [tabPage, setTabPage] = useState(0);
   const [sub, setSub] = useState(null);
-  const { title, prize, creator, submissions } = contest;
-  const isCreator = contest.creator._id === auth.user.userId;
   const [open, setOpen] = useState(false);
+  const [contest, setContest] = useState(null);
+
+  useEffect(() => {
+    const getContestDetails = async () => {
+      const { data } = await axios.get(`/contest/${match.params.id}`);
+      setContest(data);
+    };
+    getContestDetails();
+  }, []);
+
   const handleInfoClick = creator => {
     history.push(`/profile/${creator._id}`);
   };
 
   const chooseWinner = subId => {
     let url = "/contest/" + contest._id + "/subWinner?winner=" + subId;
+    const isCreator = contest.creator._id === auth.user.userId;
+
+
+    if (contest.winner) {
+      console.log('winner already chosen');
+      return;
+    }
+
     if (isCreator) {
       if (new Date() >= new Date(contest.deadline)) {
         if (!contest.winner) {
           axios
             .put(url)
             .then(res => {
               console.log(res);
-              setSub(res.data);
+              setContest(res.data.contest);
```

Comment:
This doesn't provide the same level of information as the GET /contests/:id and will result in undefined for a lot of areas. It may be better to just trigger the useEffect again